qiskit_experiments/test/pulse_backend.py

```diff
@@ -593,3 +635,210 @@ def __init__(
+    """
+
+    def __init__(
+        self,
+        qubit_frequency: float = 5e9,
+        anharmonicity: float = -0.25e9,
+        lambda_1: float = 1e9,
+        lambda_2: float = 0.8e9,
+        gamma_1: float = 1e4,
+        noise: bool = True,
+        **kwargs,
+    ):
+        """Initialise backend with hamiltonian parameters
+
+        Args:
+            qubit_frequency: Frequency of the qubit (0-1). Defaults to 5e9.
+            anharmonicity: Qubit anharmonicity $\\alpha$ = f12 - f01. Defaults to -0.25e9.
+            lambda_1: Strength of 0-1 transition. Defaults to 1e9.
+            lambda_2: Strength of 1-2 transition. Defaults to 0.8e9.
+            gamma_1: Relaxation rate (1/T1) for 1-0. Defaults to 1e4.
+            noise: Defaults to True. If True then T1 dissipation is included in the pulse-simulation.
+                The strength is given by ``gamma_1``.
+        """
+        qubit_frequency_02 = 2 * qubit_frequency + anharmonicity
+        ket0 = np.array([[1, 0, 0]]).T
+        ket1 = np.array([[0, 1, 0]]).T
+        ket2 = np.array([[0, 0, 1]]).T
+
+        sigma_m1 = ket0 @ ket1.T.conj()
+        sigma_m2 = ket1 @ ket2.T.conj()
+
+        sigma_p1 = sigma_m1.T.conj()
+        sigma_p2 = sigma_m2.T.conj()
+
+        p1 = ket1 @ ket1.T.conj()
+        p2 = ket2 @ ket2.T.conj()
+
+        ident = np.eye(3)
+
+        drift = 2 * np.pi * (qubit_frequency * p1 + qubit_frequency_02 * p2)
+        drift_2q = np.kron(drift, ident) + np.kron(ident, drift)
+
+        control = [
+            2 * np.pi * (lambda_1 * (sigma_p1 + sigma_m1) + lambda_2 * (sigma_p2 + sigma_m2))
+        ]
+        control_2q = [np.kron(ident, control[0]), np.kron(control[0], ident)]
+
+        r_frame = 2 * np.pi * qubit_frequency * (p1 + 2 * p2)
+        r_frame_2q = np.kron(r_frame, ident) + np.kron(ident, r_frame)
+
+        t1_dissipator0 = np.sqrt(gamma_1) * np.kron(ident, sigma_m1)
+        t1_dissipator1 = np.sqrt(gamma_1) * np.kron(sigma_m1, ident)
+
+        self.anharmonicity = [anharmonicity, anharmonicity]
+        self.rabi_rate_01 = [8.589, 8.589]
+        self.rabi_rate_12 = [6.876, 6.876]
+
+        if noise is True:
+            evaluation_mode = "dense_vectorized"
+            static_dissipators = [t1_dissipator0, t1_dissipator1]
+        else:
+            evaluation_mode = "dense"
+            static_dissipators = None
+
+        super().__init__(
+            static_hamiltonian=drift_2q,
+            hamiltonian_operators=control_2q,
+            static_dissipators=static_dissipators,
+            rotating_frame=r_frame_2q,
+            rwa_cutoff_freq=1.9 * qubit_frequency,
+            rwa_carrier_freqs=[qubit_frequency, qubit_frequency],
+            evaluation_mode=evaluation_mode,
+            **kwargs,
+        )
+
+        self._discriminator = [DefaultDiscriminator(), DefaultDiscriminator()]
+
+        self.subsystem_dims = (3, 3)
+
+        pulse_command = lambda qubit, name, amp: Command.from_dict(
```

Comment:
Can we reuse this setup methodology in the single-transmon case to make the code a bit lighter? Looks useful and could go in a `utils.py`.

Reply:
Yes. One option is, we could add it to the base class. If we decide to make a `utils.py` file, what other functions could we move there? I think the default discriminator,  `_get_info`, some parts of constructing the Result object,

Reply:
moved it to base class